core/domain/draft_upgrade_services.py

```diff
@@ -88,6 +95,91 @@ def try_upgrading_draft_to_exp_version(
 class DraftUpgradeUtil(python_utils.OBJECT):
     """Wrapper class that contains util functions to upgrade drafts."""
 
+    @classmethod
+    def _convert_states_v32_dict_to_v33_dict(cls, exp_id, draft_change_list):
+        """Converts draft change list from state version 32 to 33. State
+        version 33 adds image dimensions to images inside collapsible
+        blocks and tabs, for which there should be no changes to drafts.
+
+        Args:
+            exp_id: str. Exploration id.
+            draft_change_list: list(ExplorationChange). The list of
+                ExplorationChange domain objects to upgrade.
+
+        Returns:
+            list(ExplorationChange). The converted draft_change_list.
+        """
+        file_system_class = fs_services.get_entity_file_system_class()
+        exploration_fs = fs_domain.AbstractFileSystem(file_system_class(
+            feconf.ENTITY_TYPE_EXPLORATION, exp_id))
+        for i, change in enumerate(draft_change_list):
+            # Changes for html in state content.
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name == exp_domain.STATE_PROPERTY_CONTENT):
+                html_string = change.new_value['html']
+                converted_html_string = (
+                    html_validation_service.add_dims_to_img_in_complex_rte(
+                        exploration_fs, html_string))
+                draft_change_list[i].new_value[u'html'] = converted_html_string
+
+            # Changes for html in interaction answer groups.
+            if (change.cmd == exp_domain.CMD_EDIT_STATE_PROPERTY and
+                    change.property_name ==
+                    exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS):
+                updated_answer_groups = []
```

Comment:
updated_answer_group_dicts (always be careful to distinguish between the domain object and the dict).